c/aws-c-common/Makefile

```diff
@@ -0,0 +1,46 @@
+LEVEL := ../
+
+CC.Arch := 64
+
+COMMON_WARNINGS := \
+	-Wno-error=shift-count-overflow \
+	-Wno-error=overflow \
+	-Wno-error=implicit-function-declaration \
+	-Wno-error=incompatible-pointer-types \
+	-Wno-error=int-conversion \
+	-Wno-error=address \
+
+GCC_WARNINGS := \
+	-Wno-error=format= \
+	-Wno-error=discarded-qualifiers
+#	-Wno-error=builtin-declaration-mismatch
+
+CLANG_WARNINGS := \
+	-Wno-error=uninitialized
+
+ifeq ($(CC),gcc)
+COMMON_WARNINGS += $(GCC_WARNINGS)
```

Comment:
Is this and the similar statements really necessary?

The existing Makefiles already provide the variables `GCC_WARNINGS` and `CLANG_WARNINGS` and use them depending on the current compiler (cf. `Makefile-README.md`). Did this not work for you? If not, we should fix it centrally.